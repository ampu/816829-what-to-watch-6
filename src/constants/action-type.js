const ActionType = {
  SET_MOVIES_STATUS: `movies/setMoviesStatus`,
  SET_MOVIES: `movies/setMovies`,
  SET_PROMO_STATUS: `movies/setPromoStatus`,
  SET_PROMO_MOVIE: `movies/setPromoMovie`,

  SET_LOGIN_STATUS: `user/setLoginStatus`,
  SET_USER: `user/setUser`,
  SET_MY_LIST_STATUS: `user/setMyListStatus`,
  SET_MY_LIST: `user/setMyList`,

  SET_GENRE: `genres/setGenre`,
};

export default ActionType;
