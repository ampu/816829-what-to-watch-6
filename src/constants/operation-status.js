const OperationStatus = {
  UNSET: `unset`,
  PENDING: `pending`,
  RESOLVED: `resolved`,
  REJECTED: `rejected`,
  UNAUTHORIZED: `unauthorized`,
};

export default OperationStatus;
