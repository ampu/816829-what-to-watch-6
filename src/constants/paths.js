const MainPath = {
  INDEX: `/`,
  SIGN_IN: `/login`,
  MY_LIST: `/mylist`,
  MOVIE: `/films/:id`,
  ADD_REVIEW: `/films/:id/review`,
  PLAYER: `/player/:id`,
  NOT_FOUND: `/not-found`,
};

const MoviePath = {
  MOVIE_OVERVIEW: `/films/:id`,
  MOVIE_DETAILS: `/films/:id/details`,
  MOVIE_REVIEWS: `/films/:id/reviews`,
};

const MOVIE_PATHS = Array.from(new Set([MainPath.MOVIE, ...Object.values(MoviePath)]));

export {
  MainPath,
  MoviePath,
  MOVIE_PATHS,
};
