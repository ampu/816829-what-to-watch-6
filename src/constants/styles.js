const DEFAULT_BACKGROUND_STYLE = {backgroundColor: `black`};
const BORDER_BOTTOM_STYLE = {borderBottomStyle: `rgba(255, 255, 255, 0.24)`};

export {
  DEFAULT_BACKGROUND_STYLE,
  BORDER_BOTTOM_STYLE,
};
