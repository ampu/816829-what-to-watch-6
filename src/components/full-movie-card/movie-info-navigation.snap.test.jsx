import React from 'react';
import {Router, generatePath} from 'react-router-dom';
import {createMemoryHistory} from 'history';

import {render} from '@testing-library/react';

import {MainPath, MoviePath} from '../../constants/paths';

import MovieInfoNavigation from './movie-info-navigation';

describe(`MovieInfoNavigation component`, () => {
  it(`Should render according to the snapshot while at '/' url`, () => {
    const movie = {id: `fakeId`};

    const {container} = render((
      <Router history={createMemoryHistory()}>
        <MovieInfoNavigation movie={movie}/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot while at '/films/:id' url`, () => {
    const movie = {id: `fakeId`};

    const history = createMemoryHistory();
    history.push(generatePath(MainPath.MOVIE, movie));

    const {container} = render((
      <Router history={history}>
        <MovieInfoNavigation movie={movie}/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot while at '/films/:id/details' url`, () => {
    const movie = {id: `fakeId`};

    const history = createMemoryHistory();
    history.push(generatePath(MoviePath.MOVIE_DETAILS, movie));

    const {container} = render((
      <Router history={history}>
        <MovieInfoNavigation movie={movie}/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });

  it(`Should render according to the snapshot while at '/films/:id/reviews' url`, () => {
    const movie = {id: `fakeId`};

    const history = createMemoryHistory();
    history.push(generatePath(MoviePath.MOVIE_REVIEWS, movie));

    const {container} = render((
      <Router history={history}>
        <MovieInfoNavigation movie={movie}/>
      </Router>
    ));
    expect(container).toMatchSnapshot();
  });
});
