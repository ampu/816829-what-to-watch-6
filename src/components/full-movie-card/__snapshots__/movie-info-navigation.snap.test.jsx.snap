// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieInfoNavigation component Should render according to the snapshot while at '/' url 1`] = `
<div>
  <nav
    class="movie-nav movie-card__nav"
  >
    <ul
      class="movie-nav__list"
    >
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="/films/fakeId"
        >
          Overview
        </a>
      </li>
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="/films/fakeId/details"
        >
          Details
        </a>
      </li>
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="/films/fakeId/reviews"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`MovieInfoNavigation component Should render according to the snapshot while at '/films/:id' url 1`] = `
<div>
  <nav
    class="movie-nav movie-card__nav"
  >
    <ul
      class="movie-nav__list"
    >
      <li
        class="movie-nav__item movie-nav__item--active"
      >
        <a
          class="movie-nav__link"
          href="/films/fakeId"
        >
          Overview
        </a>
      </li>
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="/films/fakeId/details"
        >
          Details
        </a>
      </li>
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="/films/fakeId/reviews"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`MovieInfoNavigation component Should render according to the snapshot while at '/films/:id/details' url 1`] = `
<div>
  <nav
    class="movie-nav movie-card__nav"
  >
    <ul
      class="movie-nav__list"
    >
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="/films/fakeId"
        >
          Overview
        </a>
      </li>
      <li
        class="movie-nav__item movie-nav__item--active"
      >
        <a
          class="movie-nav__link"
          href="/films/fakeId/details"
        >
          Details
        </a>
      </li>
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="/films/fakeId/reviews"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
</div>
`;

exports[`MovieInfoNavigation component Should render according to the snapshot while at '/films/:id/reviews' url 1`] = `
<div>
  <nav
    class="movie-nav movie-card__nav"
  >
    <ul
      class="movie-nav__list"
    >
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="/films/fakeId"
        >
          Overview
        </a>
      </li>
      <li
        class="movie-nav__item"
      >
        <a
          class="movie-nav__link"
          href="/films/fakeId/details"
        >
          Details
        </a>
      </li>
      <li
        class="movie-nav__item movie-nav__item--active"
      >
        <a
          class="movie-nav__link"
          href="/films/fakeId/reviews"
        >
          Reviews
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
